### PR TITLE
ListExpr is not an lvalue anymore

### LINKED ISSUE #4340: Mypy ignores type annotation in assignment to square bracket target list
```python
from typing import Tuple, List
(x1, y1) = (1, 2)  # type: Tuple[str, int]  # E: Incompatible types in assignment (expression has type "int", variable has type "str")
[x2, y2] = (1, 2)  # type: Tuple[str, int]  # no error - bad
reveal_type(x2)  # E: Revealed type is 'builtins.int' - bad
```
Since the notation `[x1, y1]` is just another syntax for target list, the above notation should work. But even if they shouldn't, mypy should at least warn about mismatched notation.

IMHO mypy should not differentiate at all between the cases after parsing - there should be only <s>TupleType</s>TupleExpr for lvalues.

### PR DESCRIPTION
**Pull request by elazarg**
Fix #4340 by considering TupleExpr as the only lvalue sequence.

(And maybe one day it will be LvalueSeq)

### CODE REVIEW

**Comment by msullivan** [comment]
Could you add a python2 test and a test with a list lvalue and a `Tuple` annotation?

**Comment by elazarg** [comment]
Thanks! Indeed there was a Python2 bug :)

**Comment by elliott-beach** [comment]
For what it's worth, I think this is a cool change that should be merged.

File: mypy/fastparse.py [hunk 1]
```diff
@@ -960,8 +960,11 @@ def visit_Name(self, n: ast3.Name) -> NameExpr:
 
     # List(expr* elts, expr_context ctx)
     @with_line
-    def visit_List(self, n: ast3.List) -> ListExpr:
-        return ListExpr([self.visit(e) for e in n.elts])
+    def visit_List(self, n: ast3.List) -> Union[ListExpr, TupleExpr]:
+        expr_list = [self.visit(e) for e in n.elts]  # type: List[Expression]
+        if isinstance(n.ctx, ast3.Store):
```
Comment (msullivan) [inline mypy/fastparse.py]: Could you add a quick comment here and in fastparse2 explaining why?

Comment (elazarg) [reply mypy/fastparse.py]: Done

**Pull request merged**